RiotSwiftUI/Modules/Room/PollTimeline/Coordinator/PollTimelineCoordinator.swift

```diff
@@ -0,0 +1,143 @@
+// File created from SimpleUserProfileExample
+// $ createScreen.sh Room/PollTimeline PollTimeline
+/*
+ Copyright 2021 New Vector Ltd
+ 
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+ 
+ http://www.apache.org/licenses/LICENSE-2.0
+ 
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+ */
+
+import Foundation
+import UIKit
+import SwiftUI
+import MatrixSDK
+import Combine
+
+struct PollTimelineCoordinatorParameters {
+    let session: MXSession
+    let room: MXRoom
+    let pollStartEvent: MXEvent
+}
+
+@available(iOS 14.0, *)
+final class PollTimelineCoordinator: Coordinator, PollAggregatorDelegate {
+    
+    // MARK: - Properties
+    
+    // MARK: Private
+    
+    private let parameters: PollTimelineCoordinatorParameters
+    private let selectedAnswerIdentifiersSubject = PassthroughSubject<[String], Never>()
+    
+    private var pollAggregator: PollAggregator
+    private var pollTimelineViewModel: PollTimelineViewModel!
+    private var cancellables = Set<AnyCancellable>()
+    
+    // MARK: Public
+
+    // Must be used only internally
+    var childCoordinators: [Coordinator] = []
+    
+    // MARK: - Setup
+    
+    @available(iOS 14.0, *)
+    init(parameters: PollTimelineCoordinatorParameters) throws {
+        self.parameters = parameters
+        
+        try pollAggregator = PollAggregator(session: parameters.session, room: parameters.room, pollStartEvent: parameters.pollStartEvent)
+        pollAggregator.delegate = self
+        
+        pollTimelineViewModel = PollTimelineViewModel(timelinePoll: mapPoll(pollAggregator.poll))
+        pollTimelineViewModel.callback = { [weak self] result in
+            guard let self = self else { return }
+            
+            switch result {
+            case.selectedAnswerOptionsWithIdentifiers(let identifiers):
```

Comment:
```suggestion
            case .selectedAnswerOptionsWithIdentifiers(let identifiers):
```

Although you could `if case let .selectedAnswerOptionsWithIdentifiers(identifiers) = result {` seeing as you're just using the one option.

Reply:
I generally prefer to use a switch because 1) It's easier to add to it and 2) because it would warn you of unused cases when/if the enum changes.

Reply:
I think I knew that whilst I was typing tbh 😆